src/Functions/array/range.cpp

```diff
@@ -386,10 +400,27 @@ class FunctionRange : public IFunction
                             "for unsigned/signed integers up to 64 bit", getName());
         }
 
+        auto throwIfNullValue = [&](const ColumnWithTypeAndName & col)
+        {
+            if (!col.type->isNullable())
+                return;
+            const auto & nullable_col = assert_cast<const ColumnNullable &>(*col.column);
+            const auto & null_map = nullable_col.getNullMapData();
+            if (!memoryIsZero(null_map.data(), 0, null_map.size()))
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "Illegal (null) value column {} of argument of function {}", col.column->getName(), getName());
+        };
+
         ColumnPtr res;
         if (arguments.size() == 1)
         {
+            throwIfNullValue(arguments[0]);
             const auto * col = arguments[0].column.get();
+            if (arguments[0].type->isNullable())
+            {
+                const auto * nullable = checkAndGetColumn<ColumnNullable>(*arguments[0].column);
+                col = nullable->getNestedColumnPtr().get();
```

Comment:
Same here, it can be `Const(Nullable)` and `nullable` will be `nullptr`

Reply:
This case does not happen because  only one argument and useDefaultImplementationForConstants = true, const is implicitly converted to non-const. 
Test was: `SELECT range(toNullable(1));`